src/test/java/entity/PropertyTest.java

```diff
@@ -0,0 +1,164 @@
+package entity;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+/**
+ * Test class for {@link Property}.
+ * This class contains test cases to verify the behavior of the {@link Property} class methods.
+ */
+class PropertyTest {
+    static Property p;
+    static ArrayList<Integer> priceList;
+    static ArrayList<Integer> taxList;
+
+    /**
+     * Set up the test environment before each test case.
+     * Creates a new {@link Property} instance.
+     */
+    @BeforeEach
+    void setup(){
+        priceList = new ArrayList<>(List.of(1, 2, 3, 0));
+        taxList = new ArrayList<>(List.of(0, 1, 2, 3));
+
+        p = new Property(1, "Test Property", priceList, taxList);
+    }
+
+    /**
+     * Test the {@link Property#getName()} to ensure that it returns the right name
+     */
+    @Test
+    void testGetName() {
+        assertEquals("Test Property", p.getName());
+    }
+
+    /**
+     * Test the {@link Property#getBlockName()} to ensure that it returns the right name
+     */
+    @Test
+    void testGetBlockName() {
+        assertEquals("property", p.getBlockName());
+    }
+
+    /**
+     * Test the {@link Property#setName(String)} to ensure that it returns the right name
+     */
+    @Test
+    void testSetBlockName() {
+        p.setName("Modified Name");
+        assertEquals("Modified Name", p.getName());
+    }
+
+    /**
+     * Test the {@link Property#getLevel()} to ensure that the default is 0
+     */
+    @Test
+    void testGetLevel() {
+        assertEquals(0, p.getLevel());
+    }
+
+    /**
+     * Test the {@link Property#upgradeLevel()} to ensure that the level did upgrade
+     * Requires getLevel to pass
+     * Also requires getPrice and getTax to pass
+     */
+    @Test
+    void testUpgradeLevel() {
+        assertEquals(0, p.getLevel());
+        p.upgradeLevel();
+        assertEquals(1, p.getLevel());
+        assertEquals(p.getPrice(), 2);
```

Comment:
I think it's better to call with the `2` first and `p.getPrice()` next because the first argument is `expected` and second argument is `actual`, but since every test has passed and we won't change Player later, I don't think changes are needed

Reply:
I can fix that, 1 sec